src/main/java/com/incentives/piggyback/offers/serviceImpl/OfferServiceImpl.java

```diff
@@ -38,45 +42,35 @@
 	@Autowired
 	private Environment env;
 
+	@Autowired
+	private OfferRepository offerRepository;
+
+	Gson gson = new Gson();
+
 	@Override
-	public ResponseEntity<OfferDTO> updateOfferStatus(Long id, OfferDTO offers)  {
-		updateOffer(offers);
-		//PUSHING MESSAGES TO GCP
-		messagingGateway.sendToPubsub(
-				CommonUtility.stringifyEventForPublish(
-						offers.toString(),
-						Constant.OFFER_CREATED_EVENT,
-						Calendar.getInstance().getTime().toString(),
-						"",
-						Constant.OFFER_SOURCE_ID
-						));
-		return ResponseEntity.ok(offers);
+	public void offerForPartnerOrder(PartnerOrderDTO partnerOrderDTO) {
+		OfferEntity offerEntity = offerRepository.save(ObjectAdapter.generateOfferEntity(partnerOrderDTO));
+		publishOffer(offerEntity, Constant.OFFER_CREATED_EVENT);
+		List<Long> usersList = getNearbyUsers(offerEntity.getInitiatorUserId(), offerEntity.getOrderLocation().getLatitude(),
+				offerEntity.getOrderLocation().getLongitude());
+		List<UserData> usersDataList = getUsersWithInterest(usersList, partnerOrderDTO.getInterestCategories());
+		sendNotification(ObjectAdapter.generateBroadCastRequest(usersDataList, offerEntity));
 	}
+	
 
-	private OfferDTO updateOffer(OfferDTO offer) {
-		// as these are mandatory fields and should be present for update
-		if(offer.getOfferId()!=null && offer.getPartnerId()!=null && offer.getOfferCode()!=null){
-
-			if (OfferStatus.getAllStatus().contains(offer.getOfferStatus()))
-				offer.setOfferStatus(offer.getOfferStatus());
-			else
-				throw new InvalidRequestException("Invalid Status");
-
-			//date need to be updated to activate the offer
-			if(null!=offer.getOfferValidTill())
-				offer.setOfferValidTill(offer.getOfferValidTill());
-			else
-				throw new InvalidRequestException("Offer Validity date is not passed");
-
-			return offer;
-		} else {
-			throw new InvalidRequestException("Offer Id or Order Id or OfferCode cannot be null");
-		}
+	@Override
+	public void updateOfferStatus(PartnerOrderDTO partnerOrderData) {
+		List<OfferEntity> offerList = offerRepository.findByOrderId(partnerOrderData.getOrderId());
+		if (!CommonUtility.isValidList(offerList))
+			throw new InvalidRequestException("No offer available for this id");
+		OfferEntity offer = offerList.get(0);
+		offerRepository.save(ObjectAdapter.updateOfferEntity(offer, partnerOrderData));
+		publishOffer(offer, Constant.OFFER_UPDATED_EVENT);
 	}
 
 	@Override
-	public List<String> getNearbyUsers(Long userId, Double latitude, 
-			Double longitude) {
+	public List<Long> getNearbyUsers(Long userId, double latitude, 
+			double longitude) {
 		String url = env.getProperty("location.api.fetch.nearby.users");
 		HttpHeaders headers = new HttpHeaders();
 		headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
```

Comment:
Is this working? You should be generating a login request and passing Bearer token to the User API.